sdk/servicebus/Azure.Messaging.ServiceBus/tests/Receiver/ReceiverLiveTests.cs

```diff
@@ -99,6 +99,43 @@ await using (var scope = await ServiceBusScope.CreateWithQueue(enablePartitionin
             }
         }
 
+        [Test]
+        public async Task ReceiveMessagesWhenQueueEmpty()
```

Comment:
I think we also want to prove that cancelling won't increment the delivery count for any messages that were already in the Amqp library's local buffer. This may be hard to do, but maybe we can try sending a message just before we cancel?

Reply:
I can also add these tests in a follow up PR.